src/include/optimizer/projection_push_down_optimizer.h

```diff
@@ -24,6 +25,7 @@ namespace optimizer {
 class ProjectionPushDownOptimizer : public LogicalOperatorVisitor {
 public:
     void rewrite(planner::LogicalPlan* plan);
+    inline void setPathSemantic(common::PathSemantic _semantic) { this->semantic = _semantic; }
```

Comment:
I would suggest making this part of the constructor because we always expect a semantic. In this case, we will need to assign a default `PathSemantic`.

So better make it
```
ProjectionPushDownOptimizer(common::PathSemantic)
```